client/src/sagas/auth.js

```diff
@@ -1,20 +1,62 @@
-import { delay, put, takeLatest } from 'redux-saga/effects';
-import { LOGIN, LOGIN_ASYNC, REGISTER, REGISTER_ASYNC } from './types';
+import { push } from 'connected-react-router';
+import { put, takeLatest } from 'redux-saga/effects';
+import axios from 'axios';
 
-export function* loginAsync({ payload }) {
-  yield delay(1000);
-  yield put({ type: LOGIN, payload });
+import { auth } from './types';
+import { alertActions } from './actions';
+
+// logout
+export function* watchLogout() {
+  yield takeLatest(auth.LOGOUT, logout);
 }
 
-export function* registerAsync({ payload }) {
-  yield delay(1000);
-  yield put({ type: REGISTER, payload });
+export function* logout() {
+  yield put({ type: auth.reducer.LOGOUT });
+  yield put(push('/'));
 }
 
+// login
 export function* watchLoginAsync() {
-  yield takeLatest(LOGIN_ASYNC, loginAsync);
+  yield takeLatest(auth.LOGIN, loginAsync);
 }
 
+export function* loginAsync({ payload }) {
```

Comment:
📖  Why do we call this `function*`s "async"? Is that a js thing?

Reply:
It's a convention I've seen thrown around to indicate that _something_ within the `function*` is happening asynchronously (and the async behavior is sorta hidden, but it happens in L25 here)